include/nuttx/nuttx.h

```diff
@@ -48,4 +48,83 @@
 #define container_of(ptr, type, member) \
   ((type *)((uintptr_t)(ptr) - offsetof(type, member)))
 
+/* Name: max_t
+ *
+ * Description:
+ *   return maximum of two values, using the specified type
+ *
+ * Arguments:
+ *   type - data type to use
+ *   x    - first value
+ *   y    - second value
+ */
+
+#ifndef max_t
+#define max_t(type, x, y) ({               \
+        type _max1 = (x);                  \
+        type _max2 = (y);                  \
+        _max1 > _max2 ? _max1 : _max2; })
```

Comment:
this seems to be GCC specific only

Reply:
But there'a not use the keyword of typeof

Reply:
I mean not a typeof, but https://gcc.gnu.org/onlinedocs/gcc/Statement-Exprs.html

Reply:
Oh, I understand what you mean, this is indeed a problem. It can only work in the GCC compiler. If practical standard C, then it can only be implemented by functions, not macro. Do you have a good suggestion？

Reply:
Can we use c11 in nuttx? If we can, we use Generic selection: https://en.cppreference.com/w/c/language/generic

Reply:
We need to honor C89 standard since NuttX supports architectures that use C89 compilers without support for C99 and up.

I wonder what is the motivation for adding these macros? Is it for use in NuttX common code? I understand that the ordinary min() and max() macros evaluate their arguments more than once and the above max_t() evaluates each argument once; but do we really need this for use within NuttX?

Reply:
Indeed, it may not be so need